src/handlers/feature_handlers.cpp

```diff
@@ -4942,7 +4977,590 @@ void vehCarModelFeatureHandler::DrawShadow() {
+    //draw sirens
+    if (vehCarModel::SirenType < 3) {
+        if (vehCarModel::SirenType == 0 || vehCarModel::SirenType == 2) {
+            //MM2 siren
+            if (siren != nullptr && siren->getHasLights() && siren->getActive())
+            {
+                DrawSiren(model->getCarMatrix());
+            }
+        }
+        if (vehCarModel::SirenType == 1 || vehCarModel::SirenType == 2) {
+            //MM1 siren
+            Matrix44::Convert(gfxRenderState::sm_World, model->getCarMatrix());
+            gfxRenderState::m_Touched = gfxRenderState::m_Touched | 0x88;
+
+            if (siren != nullptr && siren->getActive()) {
+                bool drawLEDSiren = fmod(datTimeManager::ElapsedTime, 0.1f) > 0.05f;
+
+                if (!vehCarModel::EnableLEDSiren || drawLEDSiren) {
+                    int sirenStage = fmod(datTimeManager::ElapsedTime, 2 * vehCarModel::SirenCycle) >= vehCarModel::SirenCycle ? 1 : 0;
+                    if (sirenStage == 0 && siren0 != nullptr) {
+                        siren0->Draw(shaders);
+                    }
+                    else if (sirenStage == 1 && siren1 != nullptr) {
+                        siren1->Draw(shaders);
+                    }
+                }
+            }
+        }
+    }
+}
+
+void vehCarModelFeatureHandler::DrawHeadlights(bool rotate)
+{
+    auto model = reinterpret_cast<vehCarModel*>(this);
+
+    if (model->getHeadlight(0) == nullptr)
+        return;
+
+    bool bothLightsBroken = !(model->getEnabledElectrics(2) || model->getEnabledElectrics(3));
+    if (bothLightsBroken)
+        return;
+
+    auto carMatrix = model->getCarMatrix();
+
+    float rotationAmount = vehCarModel::HeadlightFlashingSpeed;
+
+    Vector3 camPosition = *(Vector3*)&gfxRenderState::sm_Camera->m30;
+
+    ltLight::DrawGlowBegin();
+    for (int i = 0; i < 2; i++)
+    {
+        bool isHeadlightBroken = !(model->getEnabledElectrics(i + 2));
+        if (isHeadlightBroken)
+            continue;
+
+        auto light = model->getHeadlight(i);
+        auto lightPos = model->getHeadlightPosition(i);
+
+        light->Position.Dot(lightPos, carMatrix);
+
+        if (rotate)
+        {
+            light->Direction.RotateY(datTimeManager::Seconds * rotationAmount);
+            rotationAmount *= -1.f;
+        }
+        else
+        {
+            light->Direction = Vector3(-carMatrix.m20, -carMatrix.m21, -carMatrix.m22);
+        }
+
+        if (cfgLightClippingFix.Get() == 1 || cfgLightClippingFix.Get() >= 4)
+        {
+            Vector3 vec;
+            lvlSegment segment;
+            lvlIntersection intersection;
+
+            vec.Set(light->Position);
+            vec.Subtract(camPosition);
+
+            float invGlowScale = -ltLight::GlowScale;
+
+            segment.Set(light->Position, camPosition, 0, nullptr);
+
+            if (Collide(segment, &intersection, 0, nullptr, 0x20, 0, light->Position) || Collide(segment, &intersection, 0, nullptr, 0x40, 0, light->Position))
```

Comment:
Collide uses the instanceFlags parameter (0x20 and 0x40) to check if any (not all) of those flags are set on the instance, so you should be fine combining the two calls and just passing the flags as `0x20 | 0x40`.
You should also use named constants (see previous comments).